src/properties/map_property.rs

```diff
@@ -16,28 +14,27 @@ use crate::{
 
 use super::{Property, PropertyTrait};
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct MapProperty {
     pub key_type: String,
     pub value_type: String,
     pub allocation_flags: u32,
-    #[cfg_attr(feature = "serde", serde(with = "serde_value_impl"))]
-    pub value: IndexMap<Property, Property>,
+    pub entries: Vec<(Property, Property)>,
 }
 
 impl MapProperty {
     pub fn new(
         key_type: String,
         value_type: String,
         allocation_flags: u32,
-        value: IndexMap<Property, Property>,
+        entries: Vec<(Property, Property)>,
```

Comment:
I don't think this is a good idea, because sometimes you would want to fetch by key, and iterating through the array could be quite costly, depending on the size.

Reply:
As it relates to performance, the issue is whether the hashing function used by the Map is slower than simply iterating through all of the keys in the Vec. Without proper benchmarking, we cannot say for sure which approach is more performant - however, I have not encountered any examples of MapProperty with a sufficient number of keys that would justify the use of a HashMap over a Vec in terms of performance, based on my experience using these data structures in other situations, and considering the data sizes that GVAS is designed to handle.

While I am open to analyzing this tradeoff further, as it is an API change, I am comfortable removing the commit for now to allow the other changes to proceed, and we can revisit this topic at a later time when we have more data to base the conversation on.

Reply:
This needs benchmarking, so remove the commit for now and we can get back to this conversation later with more concrete benchmarks.